lib/matplotlib/axes/_axes.py

```diff
@@ -940,12 +948,28 @@ def axline(self, xy1, xy2, **kwargs):
         axhline : for horizontal lines
         axvline : for vertical lines
         """
+        def _to_points(xy1, xy2, slope):
+            """
+            Check for a valid combination of input parameters and convert
+            to two points, if necessary.
+            """
+            if (xy2 is None and slope is None or
+                    xy2 is not None and slope is not None):
+                raise TypeError(
+                    "Exactly one of 'xy2' and 'slope' must be given")
+            if xy2 is None:
```

Comment:
I think you should support slope = inf for vertical lines.

Reply:
What’s the practical use of this?

Reply:
Also the intercept is not defined for an infinite slope

Reply:
Also the intercept is not defined for an infinite slope

Reply:
well, the point is that the parametrization here is not (slope, intercept) but (point, slope) so it's perfectly reasonable to say "draw a vertical line passing through this point" (just like one can say "draw a horizontal line passing through this point).

Reply:
That’s even more confusing then.  

That aside I don’t understand the use case for a perfectly infinite slope unless you know it’s inifinte before hand, in which case you can just call axvline.

Reply:
Support for infinite slope added. I don't know whether or not it's useful, but it's a simple straight forward extension and does not do any harm.